csharp/samples/SampleReceiver/SimpleEventProcessor.cs

```diff
@@ -0,0 +1,43 @@
+﻿namespace SampleReceiver
+{
+	using System;
+	using System.Collections.Generic;
+	using System.Text;
+	using System.Threading.Tasks;
+	using Microsoft.Azure.EventHubs;
+	using Microsoft.Azure.EventHubs.Processor;
+
+	public class SimpleEventProcessor : IEventProcessor
+	{
+		public async Task CloseAsync(PartitionContext context, CloseReason reason)
+		{
+			Console.WriteLine("Processor Shutting Down. Partition '{0}', Reason: '{1}'.", context.PartitionId, reason);
+			if (reason == CloseReason.Shutdown)
+			{
+				await context.CheckpointAsync();
```

Comment:
>await context.CheckpointAsync(); [](start = 4, length = 32)

Don't check point during shutdown. This might result in message loss. Checkpoint when messages are processed.